Singularity/Singularity/Units/GeneralUnit.cs

```diff
@@ -309,5 +313,29 @@ public void Draw(SpriteBatch spriteBatch)
                 Carrying.Get().Draw(spriteBatch);
             }
         }
+
+        public bool Die()
```

Comment:
You forgot to delete this out of the AssignedUnits list of the platform, this could be working on.

Reply:
This is happening before already - So first there's the platform dying, it's cascading through to PlatformActions and assigned units (those there get killed directly, those not get new tasks as soon as the PlatformAction is destroyed and the StructureMap notified, happening immediately after) -  so afterwards there'll be no unit assigned anymore. That's happening in PlatformBlank.cs

But I've added the unit deleting the references to the Task-referenced ...